benchmark/utils/timer_impl.hpp

```diff
@@ -195,4 +200,34 @@ class CpuTimer : public Timer {
 };
 
 
+class MpiWrappedTimer : public Timer {
+public:
+    MpiWrappedTimer(std::shared_ptr<const gko::Executor> exec,
+                    gko::experimental::mpi::communicator comm,
+                    std::shared_ptr<Timer> concrete_timer)
+        : exec_(std::move(exec)),
+          comm_(std::move(comm)),
+          concrete_timer_(std::move(concrete_timer))
+    {}
+
+protected:
+    void tic_impl() override { concrete_timer_->tic_impl(); }
+
+    double toc_impl() override
+    {
+        auto local_duration = concrete_timer_->toc_impl();
+        double duration = local_duration;
+        comm_.template all_reduce(exec_, &local_duration, &duration, 1,
```

Comment:
Is this even necessary when you provide no template parameters? It seems like the disambiguation for the parser is no longer necessary.
```suggestion
        comm_.all_reduce(exec_, &local_duration, &duration, 1,
```

Reply:
No, but sometimes CLion decides to add this...